test/parallel_api/algorithm/alg.sorting/partial_sort.pass.cpp

```diff
@@ -55,23 +55,79 @@ struct Num
 #endif
 
 template <typename Type>
-struct test_brick_partial_sort
+struct test_without_compare
+{
+    template <typename Policy, typename InputIterator>
+    typename ::std::enable_if<is_same_iterator_category<InputIterator, ::std::random_access_iterator_tag>::value &&
+                              can_use_default_less_operator<Type>::value, void>::type
+    operator()(Policy&& exec, InputIterator first, InputIterator last, InputIterator exp_first, InputIterator exp_last)
+    {
+        // The rand()%(2*n+1) encourages generation of some duplicates.
+        ::std::srand(42);
+        const ::std::size_t n = last - first;
+        for (::std::size_t k = 0; k < n; ++k)
+        {
+            first[k] = Type(rand() % (2 * n + 1));
+        }
+        ::std::copy(first, last, exp_first);
+
+        for (::std::size_t p = 0; p < n; p = p <= 16 ? p + 1 : ::std::size_t(31.415 * p))
+        {
+            auto m1 = first + p;
+            auto m2 = exp_first + p;
+
+            ::std::partial_sort(exp_first, m2, exp_last);
+#if !TEST_DPCPP_BACKEND_PRESENT
+            count_comp = 0;
+#endif
+            ::std::partial_sort(exec, first, m1, last);
+            EXPECT_EQ_N(exp_first, first, p, "wrong effect from partial_sort without predicate");
+
+#if !TEST_DPCPP_BACKEND_PRESENT
+            //checking upper bound number of comparisons; O(p*(last-first)log(middle-first)); where p - number of threads;
+            if (m1 - first > 1)
+            {
+                auto complex = ::std::ceil(n * ::std::log(float32_t(m1 - first)));
+#if TEST_TBB_BACKEND_PRESENT
+                auto p = tbb::this_task_arena::max_concurrency();
+#else
+                auto p = 1;
+#endif
+
+#ifdef PSTL_USE_DEBUG
+                if (count_comp > complex * p)
+                {
+                    ::std::cout << "complexity exceeded" << ::std::endl;
+                }
+#endif
+            }
+#endif // !TEST_DPCPP_BACKEND_PRESENT
```

Comment:
I'd suggest to remove this part (since Line86) since we have complexity check for the test below